api/xds_protocol.rst

```diff
@@ -36,43 +36,122 @@ occurs.
--  RDS: :ref:`envoy.api.v2.RouteConfiguration <envoy_api_msg_RouteConfiguration>`
--  VHDS: :ref:`envoy.api.v2.Vhds <envoy_api_msg_RouteConfiguration>`
--  CDS: :ref:`envoy.api.v2.Cluster <envoy_api_msg_Cluster>`
--  EDS: :ref:`envoy.api.v2.ClusterLoadAssignment <envoy_api_msg_ClusterLoadAssignment>`
--  SDS: :ref:`envoy.api.v2.Auth.Secret <envoy_api_msg_Auth.Secret>`
--  RTDS: :ref:`envoy.service.discovery.v2.Runtime <envoy_api_msg_service.discovery.v2.Runtime>`
+-  :ref:`envoy.api.v2.Listener <envoy_api_msg_Listener>`
+-  :ref:`envoy.api.v2.RouteConfiguration <envoy_api_msg_RouteConfiguration>`
+-  :ref:`envoy.api.v2.route.VirtualHost <envoy_api_msg_route.VirtualHost>`
+-  :ref:`envoy.api.v2.Cluster <envoy_api_msg_Cluster>`
+-  :ref:`envoy.api.v2.ClusterLoadAssignment <envoy_api_msg_ClusterLoadAssignment>`
+-  :ref:`envoy.api.v2.Auth.Secret <envoy_api_msg_Auth.Secret>`
+-  :ref:`envoy.service.discovery.v2.Runtime <envoy_api_msg_service.discovery.v2.Runtime>`
 
 The concept of `type URLs <https://developers.google.com/protocol-buffers/docs/proto3#any>`_ appears below, and takes the form
-`type.googleapis.com/<resource type>`, e.g.
-`type.googleapis.com/envoy.api.v2.Cluster` for CDS. In various
-requests from Envoy and responses by the management server, the resource
-type URL is stated.
+`type.googleapis.com/<resource type>` -- e.g., `type.googleapis.com/envoy.api.v2.Cluster` for a `Cluster` resource. In various
+requests from Envoy and responses by the management server, the resource type URL is stated.
+
+Variants of the xDS API
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Four Variants
+^^^^^^^^^^^^^
+
+There are four variants of the xDS API used via streaming gRPC, which cover all combinations of two dimensions.
+
+The first dimension is State of the World (SotW) vs. incremental. The SotW approach was the original mechanism used by xDS, in
+which the client must specify all resource names it is interested in with each request (except when making a wildcard
+request in LDS/CDS), and the server must return all resources the client has subscribed to in each request (in LDS/CDS).
+This means that if the client is already subscribing to 99 resources and wants to add an additional one, it must send a request
+with all 100 resource names, rather than just the one new one. And the server must then respond by sending all 100 resources,
+even if the 99 that were already subscribed to have not changed (in LDS/CDS). This mechanism can be a scalability limitation,
+which is why the incremental API variant was introduced. The incremental approach allows both the client and server to indicate
+only deltas relative to their previous state -- i.e., the client can say that it wants to add or remove its subscription to a
+particular resource name without resending those that have not changed, and the server can send updates only for those
+resources that have changed. For details on the incremental protocol, see :ref:`Incremental xDS <xds_protocol_incremental>`
```

Comment:
You might also want to point out that incremental supports both delta diffs _and_ _for_some_ xDS protocol a lazy load feature.